packages/flutter_tools/lib/src/asset.dart

```diff
@@ -0,0 +1,420 @@
+                     String workingDirPath: defaultWorkingDirPath,
+                     bool includeRobotoFonts: true}) async {
+    Object manifest = _loadFlutterYamlManifest(manifestPath);
+    if (manifest != null) {
+     int result = await _validateFlutterYamlManifest(manifest);
+     if (result != 0)
+       return result;
+    }
+    Map<String, dynamic> manifestDescriptor = manifest;
+    assert(manifestDescriptor != null);
+    String assetBasePath = path.dirname(path.absolute(manifestPath));
+
+    final PackageMap packageMap =
+        new PackageMap(path.join(assetBasePath, '.packages'));
+
+    Map<_Asset, List<_Asset>> assetVariants = _parseAssets(
+      packageMap,
+      manifestDescriptor,
+      assetBasePath,
+      excludeDirs: <String>[workingDirPath, path.join(assetBasePath, 'build')]
+    );
+
+    if (assetVariants == null)
+      return 1;
+
+    final bool usesMaterialDesign = (manifestDescriptor != null) &&
+        manifestDescriptor['uses-material-design'];
+
+    for (_Asset asset in assetVariants.keys) {
+      AssetBundleEntry assetEntry = _createAssetEntry(asset);
+      if (assetEntry == null)
+        return 1;
+      entries.add(assetEntry);
+
+      for (_Asset variant in assetVariants[asset]) {
+        AssetBundleEntry variantEntry = _createAssetEntry(variant);
+        if (variantEntry == null)
+          return 1;
+        entries.add(variantEntry);
+      }
+    }
+
+    List<_Asset> materialAssets = <_Asset>[];
+    if (usesMaterialDesign) {
+      materialAssets.addAll(_getMaterialAssets(_kFontSetMaterial));
+      if (includeRobotoFonts)
+        materialAssets.addAll(_getMaterialAssets(_kFontSetRoboto));
+    }
+    for (_Asset asset in materialAssets) {
+      AssetBundleEntry assetEntry = _createAssetEntry(asset);
+      if (assetEntry == null)
+        return 1;
+      entries.add(assetEntry);
+    }
+
+    entries.add(_createAssetManifest(assetVariants));
+
+    AssetBundleEntry fontManifest =
+        _createFontManifest(manifestDescriptor, usesMaterialDesign, includeRobotoFonts);
+    if (fontManifest != null)
+      entries.add(fontManifest);
+
+    // TODO(ianh): Only do the following line if we've changed packages
+    entries.add(await _obtainLicenses(packageMap, assetBasePath));
+
+    return 0;
+  }
+
+  void dump() {
+    print('Dumping AssetBundle:');
+    for (AssetBundleEntry entry in entries) {
+      print(entry.archivePath);
+    }
+  }
+}
+
+class _Asset {
+  _Asset({ this.base, String assetEntry, this.relativePath, this.source }) {
+    this._assetEntry = assetEntry;
+  }
+
+  String _assetEntry;
+
+  final String base;
+
+  /// The entry to list in the generated asset manifest.
+  String get assetEntry => _assetEntry ?? relativePath;
+
+  /// Where the resource is on disk relative to [base].
+  final String relativePath;
+
+  final String source;
+
+  File get assetFile {
+    return new File(source != null ? '$base/$source' : '$base/$relativePath');
```

Comment:
=> `new File(source != null ? '$base/$source' : '$base/$relativePath');`

?

I admit I'm kind of enamored with the expression body format.  Is your rationale that this is a long line?